lvsfunc/util.py

```diff
@@ -386,19 +336,32 @@ def get_coefs(curve: vs.TransferCharacteristics) -> Coefs:
 
 
 def check_variable_format(clip: vs.VideoNode, function: str) -> None:
-    """Check for variable format and return an error if found."""
+    """
+    Check for variable format and return an error if found.
+
+    :raises VariableFormatError:    The clip is of a variable format.
+    """
     if clip.format is None:
         raise VariableFormatError(function)
 
 
 def check_variable_resolution(clip: vs.VideoNode, function: str) -> None:
-    """Check for variable width or height and return an error if found."""
+    """
+    Check for variable width or height and return an error if found.
+
+    :raises VariableResolutionError:    The clip has a variable resolution.
+    """
     if 0 in (clip.width, clip.height):
         raise VariableResolutionError(function)
 
 
 def check_variable(clip: vs.VideoNode, function: str) -> None:
```

Comment:
For all these check_* funcs, i suggest adding a typeguard, else they're pretty much useless for typing purposes and you'll still need to assert or whatever in your main function.
https://peps.python.org/pep-0647